### PR TITLE
Operator switch

### PR DESCRIPTION
**Pull request by stevenhorsman**
Update install docs/Makefile/provisioning/e2e test to use 'upstream' CoCo payload

Fixes: #870

### CODE REVIEW

**Comment by stevenhorsman** [comment]
I've tested this on with an IBM Cloud provided cluster:
```
time="2023-05-31T04:16:36-07:00" level=info msg="Deploy the Cloud API Adaptor"
time="2023-05-31T04:16:37-07:00" level=info msg="Install the controller manager"
time="2023-05-31T04:16:48-07:00" level=trace msg="/usr/local/bin/kubectl apply -k github.com/confidential-containers/operator/config/default, output: namespace/confidential-containers-system created\ncustomresourcedefinition.apiextensions.k8s.io/ccruntimes.confidentialcontainers.org created\nserviceaccount/cc-operator-controller-manager created\nrole.rbac.authorization.k8s.io/cc-operator-leader-election-role created\nclusterrole.rbac.authorization.k8s.io/cc-operator-manager-role created\nclusterrole.rbac.authorization.k8s.io/cc-operator-metrics-reader created\nclusterrole.rbac.authorization.k8s.io/cc-operator-proxy-role created\nrolebinding.rbac.authorization.k8s.io/cc-operator-leader-election-rolebinding created\nclusterrolebinding.rbac.authorization.k8s.io/cc-operator-manager-rolebinding created\nclusterrolebinding.rbac.authorization.k8s.io/cc-operator-proxy-rolebinding created\nconfigmap/cc-operator-manager-config created\nservice/cc-operator-controller-manager-metrics-service created\ndeployment.apps/cc-operator-controller-manager created\n"
Wait for the cc-operator-controller-manager deployment be available
time="2023-05-31T04:17:04-07:00" level=info msg="Updating caa image tag with 1148ba6fb3dfda5c8453fc1a2f9649682a03018b"
time="2023-05-31T04:17:09-07:00" level=trace msg="/usr/local/bin/kubectl apply -k github.com/confidential-containers/operator/config/samples/ccruntime/peer-pods, output: ccruntime.confidentialcontainers.org/ccruntime-peer-pods created\n"
time="2023-05-31T04:17:09-07:00" level=info msg="Install the cloud-api-adaptor"
Wait for the cc-operator-daemon-install DaemonSet be available
Wait for the pod cc-operator-daemon-install-gzwn5 be ready
Wait for the cloud-api-adaptor-daemonset DaemonSet be available
Wait for the pod cloud-api-adaptor-daemonset-zvhc7 be ready
Wait for the kata-remote runtimeclass be created
=== RUN   TestCreateSimplePod
=== RUN   TestCreateSimplePod/nginx_Pod
=== RUN   TestCreateSimplePod/nginx_Pod/PodVM_is_created
time="2023-05-31T04:18:07-07:00" level=info msg="PodVM name: nginx"
time="2023-05-31T04:18:08-07:00" level=debug msg="Instance number: 0, Instance id: 0797_0225736b-b126-49e3-963a-7077de510ab9, Instance name: podvm-nginx-93bd2a3f"
time="2023-05-31T04:18:09-07:00" level=info msg="Deleting pod... nginx"
--- PASS: TestCreateSimplePod (31.81s)
    --- PASS: TestCreateSimplePod/nginx_Pod (31.81s)
        --- PASS: TestCreateSimplePod/nginx_Pod/PodVM_is_created (1.28s)
=== RUN   TestCreatePodWithConfigMap
=== RUN   TestCreatePodWithConfigMap/nginx-configmap-pod_Pod
=== RUN   TestCreatePodWithConfigMap/nginx-configmap-pod_Pod/Configmap_is_created_and_contains_data
time="2023-05-31T04:18:39-07:00" level=info msg="PodVM name: nginx-configmap-pod"
time="2023-05-31T04:18:39-07:00" level=debug msg="Instance number: 0, Instance id: 0797_69f36c71-73e4-49d7-ab4b-0d2ebe0b43ee, Instance name: podvm-nginx-configmap-pod-7f605ce9"
time="2023-05-31T04:18:41-07:00" level=info msg="Data Inside Configmap: Hello, world"
time="2023-05-31T04:18:41-07:00" level=info msg="Deleting pod... nginx-configmap-pod"
time="2023-05-31T04:18:41-07:00" level=info msg="Deleting Configmap... nginx-config"
--- PASS: TestCreatePodWithConfigMap (32.48s)
    --- PASS: TestCreatePodWithConfigMap/nginx-configmap-pod_Pod (32.48s)
        --- PASS: TestCreatePodWithConfigMap/nginx-configmap-pod_Pod/Configmap_is_created_and_contains_data (1.62s)
=== RUN   TestCreatePodWithSecret
=== RUN   TestCreatePodWithSecret/nginx-secret-pod_Pod
=== RUN   TestCreatePodWithSecret/nginx-secret-pod_Pod/Secret_has_been_created_and_contains_data
time="2023-05-31T04:19:06-07:00" level=info msg="PodVM name: nginx-secret-pod"
time="2023-05-31T04:19:07-07:00" level=debug msg="Instance number: 0, Instance id: 0797_2ac50b0e-83e8-4b1c-a614-308ecf980733, Instance name: podvm-nginx-secret-pod-4bffd95b"
time="2023-05-31T04:19:08-07:00" level=info msg="Username from secret inside pod: admin"
time="2023-05-31T04:19:09-07:00" level=info msg="Password from secret inside pod: password"
time="2023-05-31T04:19:09-07:00" level=info msg="Deleting pod... nginx-secret-pod"
time="2023-05-31T04:19:09-07:00" level=info msg="Deleting Secret... nginx-secret"
--- PASS: TestCreatePodWithSecret (28.46s)
    --- PASS: TestCreatePodWithSecret/nginx-secret-pod_Pod (28.46s)
        --- PASS: TestCreatePodWithSecret/nginx-secret-pod_Pod/Secret_has_been_created_and_contains_data (2.59s)
=== RUN   TestCreatePeerPodContainerWithExternalIPAccess
=== RUN   TestCreatePeerPodContainerWithExternalIPAccess/busybox-pod_Pod
=== RUN   TestCreatePeerPodContainerWithExternalIPAccess/busybox-pod_Pod/Peer_Pod_Container_Connected_to_External_IP
time="2023-05-31T04:19:35-07:00" level=info msg="PodVM name: busybox-pod"
time="2023-05-31T04:19:35-07:00" level=debug msg="Instance number: 0, Instance id: 0797_1861fdf6-558e-42a3-b303-49bde7c3822e, Instance name: podvm-busybox-pod-ce050d47"
time="2023-05-31T04:19:36-07:00" level=info msg="Output of ping command in busybox : PING www.google.com (142.250.187.228): 56 data bytes\n64 bytes from 142.250.187.228: seq=0 ttl=117 time=1.906 ms\n\n--- www.google.com ping statistics ---\n1 packets transmitted, 1 packets received, 0% packet loss\nround-trip min/avg/max = 1.906/1.906/1.906 ms\n"
time="2023-05-31T04:19:36-07:00" level=info msg="Deleting pod... busybox-pod"
--- PASS: TestCreatePeerPodContainerWithExternalIPAccess (26.99s)
    --- PASS: TestCreatePeerPodContainerWithExternalIPAccess/busybox-pod_Pod (26.99s)
        --- PASS: TestCreatePeerPodContainerWithExternalIPAccess/busybox-pod_Pod/Peer_Pod_Container_Connected_to_External_IP (1.48s)
PASS
ok  	github.com/confidential-containers/cloud-api-adaptor/test/e2e	192.223s
```
which worked and you can see the trace output of the `exec.Command`s,

but given that we are getting close to release and this is an installation update it would be great to get testing on one, or more of Azure, AWS, libvirt.

File: test/provisioner/provision.go [hunk 1]
```diff
@@ -131,11 +131,20 @@ func (p *CloudAPIAdaptor) Delete(ctx context.Context, cfg *envconf.Config) error
...
 	}
 
-	log.Info("Uninstall CoCo and cloud-api-adaptor")
+	log.Info("Uninstall the cloud-api-adaptor")
 	if err = p.installOverlay.Delete(ctx, cfg); err != nil {
 		return err
 	}
 
+	log.Info("Uninstall CoCo")
+	cmd := exec.Command("kubectl", "delete", "-k", "github.com/confidential-containers/operator/config/samples/ccruntime/peer-pods")
```
Comment (jtumber-ibm) [inline test/provisioner/provision.go:140]: Does the test README need to be updated to require kubectl command to be available on PATH

Comment (stevenhorsman) [reply test/provisioner/provision.go:140]: Updated in 475e95862ebca90a8c7fc51c3ee2ffe0e89035d5

**Comment by bpradipt** [comment]
I successfully deployed the operator and CAA in AKS using the latest instructions. Created a pod as well successfully.

```
$ kubectl get nodes 
NAME                                STATUS   ROLES          AGE   VERSION
aks-nodepool1-21356780-vmss000000   Ready    agent,worker   49m   v1.25.6

$ kubectl get pods -n confidential-containers-system 
NAME                                              READY   STATUS    RESTARTS   AGE
cc-operator-controller-manager-5bcb6f6d5f-9slv9   2/2     Running   0          18m
cc-operator-daemon-install-fchvk                  1/1     Running   0          18m
cc-operator-pre-install-daemon-4g2vt              1/1     Running   0          18m
cloud-api-adaptor-daemonset-z8lmb                 1/1     Running   0          10m

$ kubectl get runtimeclass
NAME          HANDLER       AGE
kata-remote   kata-remote   19m
```

File: install/README.md [hunk 2]
```diff
@@ -29,20 +29,70 @@
 
    Please refer to the instructions available in the following [doc](../webhook/docs/INSTALL.md).
 
-## Deploy cloud-api-adaptor
+## Configure and deploy CoCo and the cloud-api-adaptor
 
+- Update the `kustomization.yaml` file in `install/overlays/$(CLOUD_PROVIDER)/kustomization.yaml` with your own settings
+- Optionally [set up authenticated registry support](../docs/registries-authentication.md)
+
+## Deploy the CoCo operator, CC runtime CRD and the cloud-api-adaptor
```
Comment (bpradipt) [inline install/README.md:37]:
```suggestion
## Deploy the CoCo operator, CC runtime CRD and the cloud-api-adaptor daemonset
```

Comment (stevenhorsman) [reply install/README.md]: Updated. Thanks!

File: test/provisioner/provision.go [hunk 3]
```diff
@@ -131,11 +131,20 @@ func (p *CloudAPIAdaptor) Delete(ctx context.Context, cfg *envconf.Config) error
 		return err
 	}
 
-	log.Info("Uninstall CoCo and cloud-api-adaptor")
+	log.Info("Uninstall the cloud-api-adaptor")
 	if err = p.installOverlay.Delete(ctx, cfg); err != nil {
 		return err
 	}
 
+	log.Info("Uninstall CoCo")
```
Comment (bpradipt) [inline test/provisioner/provision.go:139]: nit: Is "Uninstall CCRuntime CRD" a better log message for this block ?

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @stevenhorsman

File: test/provisioner/provision.go (hunk 3 above)
Comment (stevenhorsman) [reply test/provisioner/provision.go:139]: Yeah, that's fair. I struggled with the log messages to find the balance between accuracy and being clear to an end user, so all help and suggestions gratefully accepted!

**Review by snir911 — APPROVED** [review]

**Pull request merged**